soc/arm/st_stm32/stm32h7/soc_m7.c

```diff
@@ -63,12 +63,11 @@ static int stm32h7_init(const struct device *arg)
 
 	SCB_EnableICache();
 
-#ifndef CONFIG_NOCACHE_MEMORY
-	if (!(SCB->CCR & SCB_CCR_DC_Msk)) {
-		SCB_EnableDCache();
+	if (!IS_ENABLED(CONFIG_NOCACHE_MEMORY)) {
+		if (!(SCB->CCR & SCB_CCR_DC_Msk)) {
+			SCB_EnableDCache();
+		}
 	}
-
-#endif /* CONFIG_NOCACHE_MEMORY */
```

Comment:
I agree this code has to be fixed. But doing it has impacts and should be done in a dedicated PR, carefully taking all the impacts into account (use of DMA for instance).

Reply:
I can easily revert this changes, they were just additional ifdefs which I replaced as requested with IS_ENABLED. As I see it this should result in the same logic as previously?

Reply:
My bad, I missed the IS_ENABLED and understood you purely removed the `CONFIG_NOCACHE_MEMORY` condition.
So it's not bad, but to avoid further confusion, it might be better to revert.
(And limit your changes to PM in this PR)

Reply:
done